src/Microsoft.Graph.Core/Requests/HttpProvider.cs

```diff
@@ -58,11 +58,21 @@ public HttpProvider(HttpClientHandler httpClientHandler, bool disposeHandler, IS
         public HttpProvider(HttpMessageHandler httpMessageHandler, bool disposeHandler, ISerializer serializer)
         {
             this.disposeHandler = disposeHandler;
-            this.httpMessageHandler = httpMessageHandler ?? new HttpClientHandler { AllowAutoRedirect = false };
+            this.httpMessageHandler = httpMessageHandler;
             this.Serializer = serializer ?? new Serializer();
 
-            GraphClientFactory.DefaultHttpHandler = () => this.httpMessageHandler;
-            this.httpClient = GraphClientFactory.Create((IAuthenticationProvider)null, "v1.0", GraphClientFactory.Global_Cloud);
+            // NOTE: Overide our pipeline when a httpMessageHandler is provided - httpMessageHandler can implement custom pipeline.
+            // This check won't be needed once we re-write the HttpProvider to work with GraphClientFactory.
+            if (this.httpMessageHandler == null)
```

Comment:
Since httpMessageHandler can be expected to be null, we should make the default value in the constructor to be null, or at least, write param doc comment indicating the behavior if httpMessageHandler is set null. Customers should have an intellisense-bsed opportunity to learn about this behavior. Otherwise, we encourage them to potentially create an unnecessary HttpMessageHandler. 

Better than what I just stated, let's add a new cstor 

```csharp
/// <summary>
/// Constructs a new <see cref="HttpProvider" with the default client middleware./>.
/// </summary>
/// <param name="disposeHandler">Whether or not to dispose the client handler on Dispose().</param>
/// <param name="serializer">A serializer for serializing and deserializing JSON objects.</param>
public HttpProvider(bool disposeHandler, ISerializer serializer) : this((HttpMessageHandler)null, disposeHandler, serializer)
{
}
```

Reply:
@MIchaelMainer HttpProvider already has a constructor overload that only accepts an `ISerializer`. The disposeHandler is used in conjunction with either `HttpMessageHandler` or `HttpClientHandler` to tell `HttpClient` whether or not it should dispose them. This will all be cleaned as part of #453.

Reply:
Thank you, missed that.